Few fixes

ryancrunchi opened the pull request:
AutoBaal : Fixed detecting hostile player as a leader.
Wakka : Fixed detecting hostile player as leader.
MFHelper : Fixed wrong leader area when receiving command.
https://github.com/kolton/d2bot-with-kolbot/issues/704
https://github.com/kolton/d2bot-with-kolbot/issues/280

DarkHorseDre commented:
> MFHelper.js : Fixed wrong leader area detected when reveiving command.
> #704
> #280
> Added town healing between commands if needed.
> 
> AutoBaal.js and Wakka.js : Fixed detecting hostile player as leader, making the bot wait in game until hostile player leave.

good work bro.  I havent looked at what you've done, but town healing between commands can cause the followers to be delayed in joining, as some commands are only seconds apart.  all players need to be present in area when killing to improve drop chance - do you cater for this?

example:  rakinishu then trist - few seconds to kill raki and enter trist, open tp, new command, follower enters tp to town, then tp to trist after command

ryancrunchi commented on d2bs/kolbot/libs/bots/AutoBaal.js:
```diff
@@ -136,7 +136,8 @@ function AutoBaal() {
 
-				if (suspect.area === destination) { // first player in our party found in destination area...
+				let playerIsHostile = getPlayerFlag(me.gid, suspect.gid, 8);
+				if (suspect.area === destination && !playerIsHostile) { // first player in our party found in destination area...
```
I find cleaner to read `playerIsHostile` than `getPlayerFlag(me.gid, suspect.gid, 8);`, also easier for future reviews and maintainability, no need comment.
Complexity of storing a variable is O(1) and performances won't be hit.

noah- replied:
seems to break with existing kolbot style, would prefer inline + comment in this case

ryancrunchi commented:
> > MFHelper.js : Fixed wrong leader area detected when reveiving command.
> > #704
> > #280
> > Added town healing between commands if needed.
> > AutoBaal.js and Wakka.js : Fixed detecting hostile player as leader, making the bot wait in game until hostile player leave.
> 
> good work bro. I havent looked at what you've done, but town healing between commands can cause the followers to be delayed in joining, as some commands are only seconds apart. all players need to be present in area when killing to improve drop chance - do you cater for this?
> 
> example: rakinishu then trist - few seconds to kill raki and enter trist, open tp, new command, follower enters tp to town, then tp to trist after command

Yes I though about when adding it. As it is a helper script, if the leader kill before helper has come, it means leader did not need help, isn't it ?
But I found sometimes that helper may enter the leader area with low life and take risk of chicken.
Also the Town.heal() function does not execute if the helper does not need healing (based on Config.HealHP and MP).

DarkHorseDre commented:
> > > MFHelper.js : Fixed wrong leader area detected when reveiving command.
> > > #704
> > > #280
> > > Added town healing between commands if needed.
> > > AutoBaal.js and Wakka.js : Fixed detecting hostile player as leader, making the bot wait in game until hostile player leave.
> > 
> > 
> > good work bro. I havent looked at what you've done, but town healing between commands can cause the followers to be delayed in joining, as some commands are only seconds apart. all players need to be present in area when killing to improve drop chance - do you cater for this?
> > example: rakinishu then trist - few seconds to kill raki and enter trist, open tp, new command, follower enters tp to town, then tp to trist after command
> 
> Yes I though about when adding it. As it is a helper script, if the leader kill before helper has come, it means leader did not need help, isn't it ?
> But I found sometimes that helper may enter the leader area with low life and take risk of chicken.
> Also the Town.heal() function does not execute if the helper does not need healing (based on Config.HealHP and MP).

It's not about needing help, it's about increasing drop chance (via number of players as bonus).  

I think the follower life scenario is moot as it shouldnt need to heal in town: player life/rejuv/belt/buffer/town/chicken config should account for these situations.  better to drink potions than to spend 10-15 seconds tp-->town-->walk-->vend-->buy-->walk-->tp

remember that in mfhelper, the helpers can spend quite a while in town depending upon the script the leader is running.  they have enough time after most runs to buy pots.  I've not had an issue with helper needing pots.  especially if their pickit is configured to pick pots too!

ryancrunchi commented:
healing to NPC is not same as doChores, it will just talk to healer and go back to portalspot.
Anyway, what do you think of being a char config `Scripts.MFHelper.healBetweenCommands = true|false;` ?

jaenster commented:
Just wondering, why a force push?

ryancrunchi commented:
> Just wondering, why a force push?

Made mistake and rebased interactively, rewriting history.

ryancrunchi commented:
> > healing is not doChores, it will just talk to healer and go back to portalspot.
> 
> true, but can still take longer to do than moving from area-->town-->newarea (kill)
> 
> > Anyway, what do you think of being a char config `Config.MFHelper.HealBetweenCommands = true|false;` for user to control this behavior ?
> 
> haha thats awesome - thank god for you guys :D

Done and added to this PR

noah- commented on d2bs/kolbot/libs/bots/MFHelper.js:
```diff
@@ -211,15 +210,17 @@ MainLoop:
 
 						if (!me.inTown && !Pather.usePortal(null, player.name)) {
 							Town.goToTown();
+							if (Config.MFHelper.HealBetweenCommands) {
```
Town.heal checks config options via "needHealing" therefore, separate config not needed for healbetweencommands

ryancrunchi replied:
What if, as @DarkHorseDre stated, you don't want the helper to heal because you decide it won't have time to, but keep the behavior of `Town.heal` in other scripts ?
The thing is, this is an optional config which default value is `false`, but when `true`, it will heal if needed, based on `needHealing`. In other scripts we don't need this kind of config because Town chores are executed between each scripts.
While writting these lines, my thought is now : Town chores are executing between each scripts, which is not the case for MFHelper which may do all the game without doing chores. What about pushing this behavior further and do
```
if (Config.MFHelper.DoChoresBetweenCommands) {
	Town.doChores();
}
```
?

noah- commented on d2bs/kolbot/libs/bots/MFHelper.js:
```diff
@@ -228,13 +229,17 @@ MainLoop:
 
 						if (!Pather.usePortal(null, player.name)) {
 							Town.goToTown();
+							if (Config.MFHelper.HealBetweenCommands) {
```
Town.heal checks config options via "needHealing" therefore, separate config not needed for healbetweencommands

ryancrunchi commented:
> Can you separate out new features to separate pull request, the fixes for checking hostile and area check for player should not be mixed with the heal changes

Doing so

> Also, all merges will be squashed as single commit, so don't need to worry about having to redo history

Yes, but my repo is fu*ked up :)

DarkHorseDre commented:
Guys I just downloaded lasted trunk and saw there is a fix in attack.js for player in wrong area, did that not fix the issue?

ps - big issue with precasting - not sure if its game or bot but confirmed the symptoms at least:  #1050
